### PR TITLE
 Migrate shell, serverNotifications and scrollHelper to ES6 + remove searchtab 

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

It removes searchTab, which is used only in the TV layout (Where tab space is already at a premium) in favor of the normal search icon in the header.

**Issues**
Part of #868

### CODE REVIEW

File: src/scripts/serverNotifications.js [hunk 1]
```diff
@@ -1,211 +1,216 @@
...
-            });
-        } else {
-            require(['alert'], function (alert) {
-                alert.default({ title: args.Header, text: args.Text });
-            });
-        }
+function displayMessage(cmd) {
+    const args = cmd.Arguments;
+    if (args.TimeoutMs) {
+        require(['toast'], function (toast) {
```
Comment (Camc314) [inline src/scripts/serverNotifications.js:18]:
```suggestion
        import('toast').then(({default: toast}) => {
```

File: src/scripts/serverNotifications.js [hunk 2]
```diff
@@ -1,211 +1,216 @@
...
+            toast({ title: args.Header, text: args.Text });
+        });
+    } else {
+        require(['alert'], function (alert) {
```
Comment (Camc314) [inline src/scripts/serverNotifications.js:22]:
```suggestion
        import('alert').then(({default: alert}) => {
```

File: src/scripts/serverNotifications.js [hunk 3]
```diff
@@ -1,211 +1,216 @@
...
+            alert.default({ title: args.Header, text: args.Text });
```
Comment (Camc314) [inline src/scripts/serverNotifications.js:23]:
```suggestion
            alert({ title: args.Header, text: args.Text });
```

File: src/scripts/serverNotifications.js (hunk 1 above)
Comment (MrTimscampi) [reply src/scripts/serverNotifications.js:18]: >  are we preferring the latter from now on?

I think it needs to be discussed with the rest of @jellyfin/web. Personally, I like consistency, so I'd say "always use parentheses". That way there's really no guessing involved.

I plan to add an ESLint rule for whatever we end up on, so please give your opinions :)

Comment (Camc314) [reply src/scripts/serverNotifications.js:18]: I agree, I think always using parentheses looks better.

File: src/scripts/scrollHelper.js [hunk 4]
```diff
@@ -1,137 +1,137 @@
...
+}
+
+function centerOnFocusHorizontal(e) {
+    centerOnFocus(e, this, true);
+}
+function centerOnFocusVertical(e) {
+    centerOnFocus(e, this, false);
+}
+
+export default {
```
Comment (dmitrylyzo) [inline src/scripts/scrollHelper.js:105]: Shouldn't `centerFocus` be exctracted and exported?

Comment (Camc314) [reply src/scripts/scrollHelper.js:105]: and `off`?

Comment (dmitrylyzo) [reply src/scripts/scrollHelper.js:105]: It is in `centerFocus`.
Should be something like this
```js
export const centerFocus = {
    on() {},
    off() {}
};

export default {
    getPosition,
    centerFocus,
    toCenter,
    toStart
};
```

Comment (Camc314) [reply src/scripts/scrollHelper.js:105]: Oh yeah my mistake

File: src/scripts/serverNotifications.js [hunk 5]
```diff
@@ -1,214 +1,216 @@
...
+function notifyApp() {
+    inputManager.notify();
+}
 
-    function notifyApp() {
-        inputManager.notify();
+function displayMessage(cmd) {
+    const args = cmd.Arguments;
+    if (args.TimeoutMs) {
+        import('toast').then((toast) => {
```
Comment (dmitrylyzo) [inline src/scripts/serverNotifications.js:18]:
```suggestion
        import('toast').then(({default: toast}) => {
```

File: src/scripts/serverNotifications.js [hunk 6]
```diff
@@ -1,214 +1,216 @@
...
+            toast({ title: args.Header, text: args.Text });
+        });
+    } else {
+        import('alert').then((alert) => {
```
Comment (dmitrylyzo) [inline src/scripts/serverNotifications.js:22]:
```suggestion
        import('alert').then(({default: alert}) => {
```

**Pull request merged**